Change to allow use of non-DF QURT drivers

jywilson opened the pull request:
Mark,

Please merge these changes with integrate3 and then submit a pull request to PX4/main when you are ready.  

Jim W.

Commit:
Defining Eagle Board specific device paths, still in progress.

Commit:
Defining Eagle Board specific device paths, still in progress.

Commit:
StubSensor template class implemented to allow registration with the DevMgr.

Commit:
StubSensor template class implemented to allow registration with the DevMgr.

Commit:
Modified the sensor module to prevent the selection of an invalid secondary/tertiary gyro if the primary gyro times out

Commit:
Modified the sensor module to prevent the selection of an invalid secondary/tertiary gyro if the primary gyro times out

Commit:
Added #ifdef needed to avoid DriverFramework initialization of QURT driver which are note yet DriverFramework compliant.

Commit:
Added #ifdef needed to avoid DriverFramework initialization of QURT driver which are note yet DriverFramework compliant.

Commit:
Removing test code.

Commit:
Removing test code.

Commit:
Removing test code.

Commit:
Removing test code.

Commit:
Removing test code.

Commit:
Removing test code.

mcharleb commented on cmake/configs/qurt_eagle_release.cmake:
```diff
@@ -70,7 +70,7 @@ set(config_module_list
 	modules/commander
 	modules/controllib
-
+	
```
Why?

jywilson replied:
Tried to remove this change, but couldn't.  Is there a git command to move a particular file to a previous commit?  I know the command to revert an entire commit.

mcharleb replied:
```
git checkout <hash-of-commit-with-old-version> cmake/configs/qurt_eagle_release.cmake
```

mcharleb commented on src/modules/sensors/sensors.cpp:
```diff
@@ -2070,6 +2070,11 @@ Sensors::task_main()
 	/* start individual sensors */
 	int ret = 0;
 
+/*
```
This change would preclude configs that did use DriverFramework drivers being used for a qurt build (there is more than one qurt build).

jywilson replied:
Good point.  The DF stub class I created should solve the problem, but we need to get DF to successfully build with the QURT config.

jywilson replied:
Let me know if you will be working on the above problem.  If so, when you are done I will replace my changes to the driver that use the stub class.

mcharleb commented on src/modules/sensors/sensors.cpp:
```diff
@@ -2215,15 +2221,21 @@ Sensors::task_main()
+				if (_gyro_sub[1] != -1) {
+					warnx("using secondary gyro");
+					fds[0].fd = _gyro_sub[1];
+				}
```
There should be some indication that main, tertiary and secondary failed or were not available.

jywilson replied:
Good point.  I will make the change.

mcharleb commented on src/platforms/qurt/include/qurt_log.h:
```diff
@@ -48,6 +48,8 @@ __EXPORT extern uint64_t hrt_absolute_time(void);
 // declaration to make the compiler happy.  This symbol is part of the adsp static image.
 void HAP_debug(const char *msg, int level, const char *filename, int line);
 
```
There should be a #pragma once added to the file, not the following check.

Done.

Commit:
Removing changes made inadvertantly to the make file.

Commit:
Removing changes made inadvertantly to the make file.

Commit:
Adding additional warnings to indicate that the secondary and tertiary gyros are unavailable.

Commit:
Adding additional warnings to indicate that the secondary and tertiary gyros are unavailable.

Commit:
Adding #pragma once to header file.

Commit:
Merge commit '8d25ea554cdc114bdfdb62c9eeebfda150349a6c' into integration3_jim

julianoes committed:
df_imu: remove debug relict

julianoes committed:
df_imu: added gyro publishing, cleanup

julianoes committed:
DriverFramework: updated submodule

julianoes committed:
uORB: improve console output and status

julianoes committed:
POSIX main: always add newline after a command

julianoes committed:
apps.h_in: fixed what is presumably a typo

julianoes committed:
muorb: don't start muorb with stop command

Previously using 'muorb stop' would in fact start at least part of the
muorb which is not what it's expected to do.

julianoes committed:
POSIX main: set exit flag and exit gracefully

On Control+C, 'muorb stop' is now always executed, however this
shouldn't break POSIX SITL where the command is just not available.

julianoes committed:
POSIX main: get rid of some leftover printfs

julianoes committed:
sensors: header comment changes

julianoes committed:
uORB: fix off-by-one bug in topic count

julianoes committed:
sensors: don't just blindly switch to another gyro

Failover is good but failover to garbage doesn't help.

julianoes committed:
sensors: HACK to stop init calls for sensors

julianoes committed:
Rename DfImu to DfMpu9250Wrapper

julianoes committed:
attitude_estimator_q: be specific about the error

mcharleb committed:
Set stack size to minimum valid size if less requested

The stack size cannot be less than PTHREAD_STACK_MIN.

Signed-off-by: Mark Charlebois <charlebm@gmail.com>

mcharleb committed:
FIxed travis build to resolve link symbols

Signed-off-by: Mark Charlebois <charlebm@gmail.com>

Commit:
Debug statements included in the code.

Commit:
Debug statements included in the code.

Commit:
Removing unneeded debug statements.

Commit:
Removing unneeded debug statements.

julianoes committed:
BMP280: added the baro driver

julianoes committed:
DriverFramework: updated submodule

julianoes committed:
px4_qurt_tasks: fix header

julianoes committed:
px4_qurt_tasks: fix hang because of absolute time

The timeout was triggered using absolute time instead of a delay in
usec. This lead to the system hanging. With the fix it continues after
the timeout, however, the rates still don't seem right.

julianoes committed:
px4_qurt_tasks: fix timeout calculation

It was wrong to subtract seconds and nanoseconds from each other. This
will wrap, therefore use ts_to_abstime and do the subtraction
afterwards.

julianoes committed:
POSIX: get px4_poll right

Like this the timeouts are going through instead of hanging everything.

julianoes committed:
QURT: use drv_hrt.c from POSIX

mcharleb committed:
Fixed terminal when exiting mainapp

mainapp was not restoring the original terminal setttings on exit

Signed-off-by: Mark Charlebois <charlebm@gmail.com>

Commit:
This code is flyable, but a few problems exist which cause values in the .config files to be overwritten by the defaults in the .XML file.

Commit:
Resolving merge conflicts in qurt_log.h.

Commit:
Resolved shared memory parameter problems and removed additional debug code.

Commit:
Fixed problem causing a failure to obtain the shared memory lock on the AppsProc.

Commit:
Added build specific sensor initialization.
